### PR TITLE
[Bug] Update Prebuilt Detection Rules Release Process

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/3374

## Summary
This pull request fixes the Detection Rules release workflow and related CLI commands used. Below is a list of issues that were addressed and have more detailed information in the related issue.

- Automatic Locked Version Workflow Commit Hash Identification
- Incorrect use of branch version for determining previous release packages from EPR
- Add historical rules not default since the dropped support for 8.7 and previous 
- Integrations PR labeling and description updates

```[tasklist]
- [x] Fix/Update Release Meta Template in TRADE Repository
- [x] Review release-fleet workflow for updates and bugs
- [x] Adjust fleet package building to add historical rules by default
- [x] Adjust `packages.yml` in 8.10 branch so `kibana.version` is `8.10.1`
- [x] Update documentation for step to ask for backport branch in integrations if not available
- [x] Update `bump-pkg-versions` command to address schema failures
```

## Testing
Testing will be a big part of these updates. The importance of testing it to ensure these important steps work as intended:

- Locked Versions workflow commit is properly obtained and used after checking out protected branches
- When `bump-pkg-versions` CLI command is used, `packages.yml` reflects the accurate results; only distinguishable when integrations PR is created as it will reflect in the manifest

Example run: https://github.com/elastic/detection-rules/actions/runs/7791122549

Release Workflow Test:
1. Detection Rules repo > Actions
2. Select `release-fleet` > Run workflow
3. Change "User workflow from" from `main` to `bug-release-process-updates`
4. Leave default options
5. Select "run workflow"
6. Monitor workflow progress
7. Under `Build package and create PR to integrations` job > `Build package and create PR to integrations`; ensure that it checks out a version lock commit hash as shown below

Screenshot 2024-02-05 at 4 35 15 PM

9. Note that the `Build release package` step is expected to fail. This is because we are checking out a commit hash that is PRIOR to this PR. As a result, when the package builds, it will incorrectly add `release` to the manifest which is fixed [here](https://github.com/elastic/detection-rules/blob/c44eef633eb67f65527ecb2f4b4891ab4a5b4a9d/detection_rules/devtools.py#L230-L231).

### CODE REVIEW

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -192,7 +194,7 @@ def bump_versions(major_release: bool, minor_release: bool, patch_release: bool,
     """Bump the versions"""
 
     pkg_data = load_etc_dump('packages.yml')['package']
-    kibana_ver = Version.parse(pkg_data["name"], optional_minor_and_patch=True)
+    kibana_ver = Version.parse(pkg_data['name'], optional_minor_and_patch=True)
```
Comment (terrancedejesus) [inline detection_rules/devtools.py:197]:
```suggestion
    kibana_ver = Version.parse(pkg_data["name"], optional_minor_and_patch=True)
```

**Review by eric-forte-elastic — COMMENTED** [review]
NIt:

Since we are now defining `registry_data = config['registry_data']`

I think we can reuse it in places like the following to make;
```python
docs = IntegrationSecurityDocsMDX(config['registry_data']['version'], Path(f'releases/{config["name"]}-docs'),
                                          True, historical_rules, package, note=update_message
```
```python
 
 docs = IntegrationSecurityDocsMDX(registry_data['version'], Path(f'releases/{config["name"]}-docs'),
                                          True, historical_rules, package, note=update_message)
 
 ```

**Comment by terrancedejesus** [comment]
>Since we are now defining registry_data = config['registry_data']. I think we can reuse it in places like the following to make;

Good point, updated.

**Comment by terrancedejesus** [comment]
- Check that branch workflow ran from is consistent with checkout

**Review by eric-forte-elastic — APPROVED** [review]
Great peer review and demo/testing meeting. LGTM :+1:

File: .github/workflows/release-fleet.yml [hunk 2]
```diff
@@ -28,162 +28,129 @@ on:
...
+                core.setFailed('Forbidden branch')
+              }
+        - name: Checkout detection-rules
+          uses: actions/checkout@v3
+          with:
+            path: detection-rules
+            fetch-depth: 0
+
+        - name: Extract version lock commit hash
+          id: extract_commit_hash
```
Comment (Mikaayenson) [inline .github/workflows/release-fleet.yml:55]: Do you want to make any following steps still dependent on this step?

Comment (terrancedejesus) [reply .github/workflows/release-fleet.yml]: Updated to remove ID.

**Review by shashank-elastic — APPROVED** [review]

**Pull request merged**